cookbook/routing/extending_routing.rst

```diff
@@ -0,0 +1,15 @@
+Extending the Routing System
```

Comment:
this is the overview of the subsection on extending the routing, trying to give an overview.